src/devdata/extras.py

```diff
@@ -155,6 +155,19 @@ def check_simple_value(mapping: Dict[str, str], *, key: str) -> str:
                 name = check_simple_value(sequence, key="sequencename")
                 data_type = check_simple_value(sequence, key="data_type")
 
+                # Support reset modes which don't drop the database. At some
+                # point it might be nice to be able to hook into the reset mode
+                # to remove sequences too, however that's likely complicated and
+                # it's easy enough to handle here.
+                #
+                # Sequences don't nicely fit into one of just schema or data,
+                # they're somewhat inherently both. Given that Django's
+                # "loaddata" over-writes existing rows in tables, it seems
+                # reasonable to do something similar for sequences -- even if
+                # that means we actually drop the sequence and fully re-create
+                # it.
+                cursor.execute(f"DROP SEQUENCE IF EXISTS {name}")
```

Comment:
Should we only do this for reset modes that don't drop the database? Should we add a hook in the reset mode for resetting the sequence?

Reply:
Maaybe. This is probably the part of this PR I'm least happy with. I considered hooking into the reset modes here, however it's not clear how that would work especially given that this extra is Postgres specific while the reset modes are (I believe mostly) sufficiently general to any of Django's supported backends[^1].

To clarify: is your concern here with the possible outcomes under some combination(s) of existing state & reset mode, or with the approach to solving the problem? (or both?)

Running through my thinking on the interactions with each mode:
- `drop-database`: nothing needed (sequence already gone)
- `drop-tables`: should probably drop the sequence (as this code does)
- `none`: unclear; leaving an existing sequence in place would always error during import, while an import over an empty database would just work. In contrast, I'm expecting (though haven't tested and should) that `none` against an existing matching database with the right schema (and migrations table) would import cleanly (modulo conflicts with the existing data).

I _think_ that the behaviour this code achieves is approximately the same as equivalent schema/data imports and probably most likely what the user wants (regardless of mode). Agree it doesn't feel like the cleanest way to achieve that though!

[^1]: I suspect that some of the `IF EXISTS` usage which is now part of the reset modes but was already present may in fact be Postgres specific, though this was an existing limitation.

Reply:
I've added some docs around this as part of 729a19f3a0430df73e75b022f77a4373746da226 (which also adds more docs for the reset modes).